### PR TITLE
Unified deploy method, required by fuel-2.2

### PR DESCRIPTION
**Pull request by dmi-try**

### CODE REVIEW

File: lib/astute/deployment_engine.rb [hunk 1]
```diff
@@ -59,18 +59,7 @@ def attrs_multinode(nodes, attrs)
...
-
-      compute_nodes = nodes.select {|n| n['role'] == 'compute'}
-      Astute.logger.info "Starting deployment of computes"
-      deploy_piece(compute_nodes, attrs)
-
-      other_nodes = nodes - ctrl_nodes - compute_nodes
-      Astute.logger.info "Starting deployment of other nodes"
-      deploy_piece(other_nodes, attrs)
-      return
+      deploy_ha_full(nodes, attrs)
```
Comment (AlexanderPavlenko) [inline lib/astute/deployment_engine.rb:62]: So, `deploy_multinode` is an alias for `deploy_ha_full`?

Comment (dmi-try) [inline lib/astute/deployment_engine.rb:62]: all deploy_\* methods are aliases to the deploy_ha_full. After merge with Victor's branch all these methods should be removed.

**Comment by AlexanderPavlenko** [comment]
+1

File: lib/astute/deployment_engine.rb [hunk 2]
```diff
@@ -87,6 +76,17 @@ def attrs_ha(nodes, attrs)
...
       end
 
+      attrs['nodes'] = ctrl_nodes.map do |n|
+        {
+          'name'             => n['fqdn'].split(/\./)[0],
+          'role'             => 'controller',
+          'internal_address' => n['network_data'].select {|nd| nd['name'] == 'management'}[0]['ip'].split(/\//)[0],
+          'public_address'   => n['network_data'].select {|nd| nd['name'] == 'public'}[0]['ip'].split(/\//)[0],
+          'mountpoints'      => "1 1\n2 2",
+          'storage_local_net_ip' => n['network_data'].select {|nd| nd['name'] == 'management'}[0]['ip'].split(/\//)[0],
```
Comment (mihgen) [inline lib/astute/deployment_engine.rb]: Why do we use management network for storage, should not we use storage network?

File: lib/astute/deployment_engine.rb [hunk 3]
```diff
@@ -95,101 +95,40 @@ def attrs_ha(nodes, attrs)
...
       storage_nodes = nodes.select {|n| n['role'] == 'storage'}
       proxy_nodes = nodes.select {|n| n['role'] == 'swift-proxy'}
-      other_nodes = nodes - ctrl_nodes - compute_nodes - quantum_nodes - storage_nodes -proxy_nodes
-
-      Astute.logger.info "Starting deployment of all controllers one by one"
-      ctrl_nodes.each {|n| deploy_piece([n], attrs, retries=0)}
-
-      Astute.logger.info "Starting deployment of 1st controller again"
-      deploy_piece(ctrl_nodes[0..0], attrs, retries=0)
+      primary_proxy_nodes = nodes.select {|n| n['role'] == 'primary-swift-proxy'}
```
Comment (mihgen) [inline lib/astute/deployment_engine.rb:117]: Do we have such role in nailgun?

Comment (dmi-try) [inline lib/astute/deployment_engine.rb:117]: No, it is for fuel only.

**Pull request merged**